charts/partials/templates/_proxy.tpl

```diff
@@ -29,7 +29,7 @@ env:
   value: {{.Values.proxy.requireTLSOnInboundPorts | quote}}
 {{ end -}}
 - name: LINKERD2_PROXY_LOG
-  value: {{.Values.proxy.logLevel | quote}}
+  value: "{{.Values.proxy.logLevel}}{{ if not (eq .Values.proxy.logHTTPHeaders "insecure") }},linkerd_proxy_http::client[{headers}]=off{{ end }}"
```

Comment:
I think it still is a good practice to leave the `quote` function in.

Reply:
Can you explain? it was my impression that the quote function just adds the quotation marks.  Using quote here is difficult because its a pipeline function and we're constructing the string in the template.